kde-settings-qubes.spec.in

```diff
@@ -117,6 +117,16 @@ sed -i 's|XDG_MENU_PREFIX="kf5-"|XDG_MENU_PREFIX="kf5-qubes-"|g' /etc/xdg/plasma
 %settings_replace /usr/share/plasma/look-and-feel/org.fedoraproject.fedora.desktop/contents/plasmoidsetupscripts/org.kde.plasma.kickerdash.js.qubes
 %settings_replace /usr/share/plasma/look-and-feel/org.fedoraproject.fedora.desktop/contents/plasmoidsetupscripts/org.kde.plasma.kickoff.js.qubes
 
+%triggerin -- plasma-desktop
+default_layout="/usr/share/plasma/layout-templates/org.kde.plasma.desktop.defaultPanel/contents/layout.js"
+if ! grep -q qubesMenu "$default_layout"; then
+    cp -a "$default_layout" "$default_layout.qubes-orig"
+    sed -i \
+        -e "s#.*org.kde.plasma.kickoff.*#var qubesMenu = panel.addWidget('org.kde.plasma.quicklaunch')\nqubesMenu.currentConfigGroup = ['General']\nqubesMenu.writeConfig('launcherUrls', ['file:///usr/share/applications/open-qubes-app-menu.desktop'])#" \
+        -e "s/^kickoff/qubesMenu/" \
+        "$default_layout"
+fi
```

Comment:
Using `sed` on a JavaScript file???  On the one hand this is a disgusting hack, but on the other hand the proper way to do this would be to use a JavaScript parser and I don’t know of one that is suitable for this.

Reply:
This is an alternative to shipping own copy of the whole file, not to a full parser. I could use a patch, but honestly, it doesn't matter that much in this place.

Reply:
A patch would make it _much_ easier to understand what is going on here.  That said I think shipping the full file would be best, especially for R4.2 users who are still getting dom0 updates from Fedora.

Reply:
> still getting dom0 updates from Fedora.

That's exactly the reason why I do _not_ want to ship the whole file - I only want to replace the menu, while keeping upstream layout of the rest of the panel.